sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubBufferedProducerClient.cs

```diff
@@ -2314,16 +2380,36 @@ private async Task UpdatePartitionInformation(CancellationToken cancellationToke
           ((waitTime == Timeout.InfiniteTimeSpan) || (waitTime > minimumAllowedWaitTime));
 
         /// <summary>
-        ///   Calculates the amount of delay to apply, ensuring that the remaining time allotted
-        ///   supersedes the delay amount, if not enough time remains for the full delay.
+        ///   Calculates the amount of delay to apply when building a batch, ensuring that the remaining time
+        ///   allotted supersedes the delay amount, if not enough time remains for the full delay.
         /// </summary>
         /// <param name="remainingTime">The amount of allotted time remaining.</param>
         /// <param name="delayInterval">The desired delay interval.</param>
         ///
         /// <returns>The amount of delay to apply.</returns>
         ///
-        private static TimeSpan CalculateDelay(TimeSpan remainingTime,
-                                               TimeSpan delayInterval) => ((remainingTime != Timeout.InfiniteTimeSpan) && (remainingTime < delayInterval)) ? remainingTime : delayInterval;
+        private static TimeSpan CalculateBatchingDelay(TimeSpan remainingTime,
+                                                       TimeSpan delayInterval) => ((remainingTime != Timeout.InfiniteTimeSpan) && (remainingTime < delayInterval)) ? remainingTime : delayInterval;
+
+        /// <summary>
+        ///   Calculates the amount of delay to apply between checks when no events are available
+        ///   in the buffers for any partition, gradually increasing for each empty check until it
+        ///   reaches the maximum limit.
+        /// </summary>
+        ///
+        /// <param name="consecutiveEmptyBufferCount">The consecutive number of times that the buffers have been empty when checked.</param>
+        /// <param name="startingEmptyBufferDelaySeconds">The number of seconds to use as the starting point for the delay.</param>
+        /// <param name="maximumBufferDelaySeconds">The maximum number of seconds to allow as a delay.</param>
+        ///
+        /// <returns>The amount of delay to apply before checking the buffers for available events.</returns>
+        ///
+        private static TimeSpan CalculateEmptyBufferDelay(int consecutiveEmptyBufferCount,
+                                                          double startingEmptyBufferDelaySeconds,
+                                                          double maximumBufferDelaySeconds)
+        {
+            var delay = (Math.Pow(1.4, consecutiveEmptyBufferCount) * startingEmptyBufferDelaySeconds);
```

Comment:
Just curious why we picked `Math.Pow(1.4, consecutiveEmptyBufferCount)` as opposed to something more straightforward like `consecutiveEmptyBufferCount * 2`.